lib/ansible/modules/cloud/google/gcp_sql_instance.py

```diff
@@ -0,0 +1,967 @@
+                              the IP. In CIDR notation, also known as 'slash' notation (e.g. 192.168.100.0/24).
+                        returned: success
+                        type: complex
+                        contains:
+                            expiration_time:
+                                description:
+                                    - The time when this access control entry expires in RFC 3339 format, for example
+                                      2012-11-15T16:19:00.094Z.
+                                returned: success
+                                type: str
+                            name:
+                                description:
+                                    - An optional label to identify this entry.
+                                returned: success
+                                type: str
+                            value:
+                                description:
+                                    - The whitelisted value for the access control list. For example, to grant access
+                                      to a client from an external IP (IPv4 or IPv6) address or subnet, use that address
+                                      or subnet here.
+                                returned: success
+                                type: str
+                    require_ssl:
+                        description:
+                            - Whether the mysqld should default to 'REQUIRE X509' for users connecting over IP.
+                        returned: success
+                        type: bool
+            tier:
+                description:
+                    - The tier or machine type for this instance, for example db-n1-standard-1. For MySQL
+                      instances, this field determines whether the instance is Second Generation (recommended)
+                      or First Generation.
+                returned: success
+                type: str
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.module_utils.gcp_utils import navigate_hash, GcpSession, GcpModule, GcpRequest, remove_nones_from_dict, replace_resource_dict
+import json
+import time
+
+################################################################################
+# Main
+################################################################################
+
+
+def main():
+    """Main function"""
+
+    module = GcpModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            backend_type=dict(type='str', choices=['FIRST_GEN', 'SECOND_GEN', 'EXTERNAL']),
+            connection_name=dict(type='str'),
+            database_version=dict(type='str', choices=['MYSQL_5_5', 'MYSQL_5_6', 'MYSQL_5_7', 'POSTGRES_9_6']),
```

Comment:
Is there a dynamic way for these choices to be discovered, maybe a preflight call to a list of available DB's, so that when Postgres 9.7 or MySQL 5.8 is available, the module doesn't need to be changed?

Reply:
You would think there would be. We don't seem to have an API to return it though.